app/src/main/java/com/aptoide/uploader/account/network/RetrofitAccountService.java

```
@@ -42,27 +46,76 @@ public RetrofitAccountService(ServiceV3 serviceV3, ServiceV7 serviceV7,
     this.authenticationProvider = authenticationProvider;
   }
 
-  @Override public Single<AptoideAccount> getAccount(String username, String password) {
+  @Override public Single<Account> getAccount(String username, String password) {
     return authenticationProvider.getAccessToken(username, password)
         .flatMap(accessToken -> serviceV7.getUserInfo(
             new AccountRequestBody(Collections.singletonList("meta"), accessToken))
             .singleOrError())
         .flatMap(response -> {
           if (response.isSuccessful() && response.body()
               .isOk()) {
-            return Single.just(mapper.map(response.body()));
+            return Single.just(mapper.map(response.body(), BaseAccount.LoginType.APTOIDE));
           }
           return Single.error(new IllegalStateException(response.message()));
         });
   }
 
-  @Override
-  public Single<AptoideAccount> createAccount(String email, String password, String storeName) {
+  @Override public Single<Account> getAccountOAuth(String email, String token, String authMode) {
+    return authenticationProvider.getAccessTokenOAuth(email, token, authMode)
+        .flatMap(accessToken -> serviceV7.getUserInfo(
+            new AccountRequestBody(Collections.singletonList("meta"), accessToken))
+            .singleOrError())
+        .flatMap(response -> {
+          if (response.isSuccessful() && response.body()
+              .isOk()) {
+            BaseAccount.LoginType loginType;
+            if (authMode.equals("facebook_uploader")) {
+              loginType = BaseAccount.LoginType.FACEBOOK;
+            } else {
+              loginType = BaseAccount.LoginType.GOOGLE;
```

here we should check if authMode is google and not having a default to it. Could hurt in the future if we add different kind of social login (like twitter or github for instance). The default should throw a runtime exception with a specific message)